src/lib/dynamic_help_config.ts

```diff
@@ -0,0 +1,199 @@
+
+export type GuestArrivalHelpSetItem =
+    | "right-nav-help"
+    | "settings-button-help"
+    | "username-change-help"
+    | "profile-button-username-help"
+    | "profile-page-username-help";
+
+export type HelperSetItem = NewUserHelpSetItem | GuestArrivalHelpSetItem;
+
+export type DynamicHelpSetItemSchema = {
+    // indicates if the item should be visible at the component level.  The set has to be enabled as well.
+    show_item: boolean;
+    // indicates if the item should be turned on when the whole set is turned on.
+    // (useful for resetting sets that sequence their messages)
+    set_initially: boolean;
+};
+
+export type DynamicHelpSetSchema = {
+    show_set: boolean;
+    set_title: string;
+    items: { [item_name in HelperSetItem]?: DynamicHelpSetItemSchema };
+};
+
+export type DynamicHelpSchema = {
+    [set_name in DynamicHelpSet]: DynamicHelpSetSchema;
+};
+
+// This is currently used to define what items are in a help set,
+// so we can make sure we turn them all on in `showHelpSet()` and related
+export const DEFAULT_DYNAMIC_HELP_CONFIG: DynamicHelpSchema = {
+    "new-user-help-set": {
+        show_set: false,
+        set_title: pgettext(
+            "Label for the settings controlling help for newly registered users",
+            "New User Help Set",
+        ),
+        items: {
+            "new-user-welcome": { show_item: false, set_initially: true },
+            "new-user-verify-email-in-settings": { show_item: false, set_initially: true },
+        },
+    },
+    "guest-arrival-help-set": {
+        show_set: false,
+        set_title: pgettext(
+            "Label for the settings controlling help for arriving guests",
+            "Guest Arrival Help Set",
+        ),
+        items: {
+            "right-nav-help": { show_item: false, set_initially: true },
+            "settings-button-help": { show_item: false, set_initially: true },
+            "username-change-help": { show_item: false, set_initially: false },
+            "profile-button-username-help": { show_item: false, set_initially: false },
+            "profile-page-username-help": { show_item: false, set_initially: false },
+        },
+    },
+};
+
+export function setIsVisible(set_name: DynamicHelpSet): boolean {
+    return data.get(`dynamic-help.${set_name}`, DEFAULT_DYNAMIC_HELP_CONFIG[set_name]).show_set;
+}
+
+export function allItemsVisible(set_name: DynamicHelpSet): boolean {
+    const config = data.get(`dynamic-help.${set_name}`, DEFAULT_DYNAMIC_HELP_CONFIG[set_name]);
+
+    return Object.keys(config.items).reduce(
+        (prev, current) => prev && config.items[current].show_item,
+        config.show_set,
+    );
+}
+
+export function initialItemsVisible(set_name: DynamicHelpSet): boolean {
+    const config = data.get(`dynamic-help.${set_name}`, DEFAULT_DYNAMIC_HELP_CONFIG[set_name]);
+
+    return Object.keys(config.items).reduce(
```

Comment:
`Object.values`, `Array.prototype.every` and `Array.prototype.some` can be used to make these statements a little simpler:

```
// All
Object.values(config.items).every((item) => item.show_item);
// Initial
Object.values(config.items).every((item) => !item.set_initially || item.show_item);
// Some
Object.values(config.items).some((item) => item.show_item);
```